### PR TITLE
initial ctc/rnn-t standalone example setup

### PR DESCRIPTION
**Pull request by JackTemaki**
This is an example setup for my CTC/RNN-T pipeline. Currently only LibriSpeech CTC is added, with dev-other 6.7% and test-other 7.0%.

For the review I will not accept substantial changes or new jobs, but this more about questions, details and (moderate) wishes. This is just an example setup, and should not be imported from, but use as a starting point reference which can on demand be copied into the user folder for a new personal setup.

Please also note the example_settings.py, which might be a nicer reference for future Sisyphus and cluster settings.

### CODE REVIEW

File: example_setups/librispeech/ctc_rnnt_standalone_2024/config.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    post_config = {
+        "num_workers_per_gpu": 4,
+    }
+
+    base_config = {
+        #############
+        "batch_size": 2000 * 16000,
+        "max_seqs": 240,
+        #############
+        "forward": training_datasets.prior.as_returnn_opts(),
```
Comment (Atticus1806) [inline example_setups/librispeech/ctc_rnnt_standalone_2024/config.py:112]:
```suggestion
        "forward": copy.deepcopy(training_datasets.prior.as_returnn_opts()),
```

File: example_setups/librispeech/ctc_rnnt_standalone_2024/experiments/ctc_bpe/baseline.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        conv_dropout=0.1,
+        ff_dropout=0.1,
+        mhsa_dropout=0.1,
+        conv_kernel_size=31,
+        final_dropout=0.1,
+        specauc_start_epoch=11,  # BPE does not converge otherwise
+    )
+
+    train_config_24gbgpu_amp = {
+        "optimizer": {"class": "adamw", "epsilon": 1e-16, "weight_decay": 1e-3},
```
[the remark this answers is not in the document]
Comment (JackTemaki) [reply example_setups/librispeech/ctc_rnnt_standalone_2024/experiments/ctc_bpe/baseline.py:193]: For LS100 there were some improvements, for LS960 I do not know...

File: example_setups/librispeech/ctc_rnnt_standalone_2024/experiments/ctc_bpe/baseline.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+from ...default_tools import RETURNN_EXE, MINI_RETURNN_ROOT
+from ...lm import get_4gram_binary_lm
+from ...pipeline import training, prepare_asr_model, search, ASRModel
+
+
+def bpe_ls960_1023_base():
+    prefix_name = "experiments/librispeech/ctc_rnnt_standalone_2024/ls960_ctc_bpe_5k"
+
+    train_settings = DatasetSettings(
+        preemphasis=0.97,  # TODO: Check if this is really useful
```
[the remark this answers is not in the document]
Comment (JackTemaki) [reply example_setups/librispeech/ctc_rnnt_standalone_2024/experiments/ctc_bpe/baseline.py:23]: Good to know, many of my former attention baselines did not have it so I considered this an error actually...

[the remark this answers is not in the document]
Comment (michelwi) [reply example_setups/librispeech/ctc_rnnt_standalone_2024/experiments/ctc_bpe/baseline.py:23]: In our case if was a hybrid model. That could make a difference and attention does not need it.. 🤷

[the remark this answers is not in the document]
Comment (JackTemaki) [reply example_setups/librispeech/ctc_rnnt_standalone_2024/experiments/ctc_bpe/baseline.py:23]: I will still keep it here until proven to hurt / not be necessary

**Pull request merged**